clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -511,7 +511,7 @@ public void logUnused() {
         Map<String, ConfigProvider> configProviderInstances = new HashMap<>();
         for (Map.Entry<String, String> entry : providerMap.entrySet()) {
             try {
-                String prefix = CONFIG_PROVIDERS_CONFIG + "." + entry.getKey() + ".";
+                String prefix = CONFIG_PROVIDERS_CONFIG + "." + entry.getKey() + ".param.";
```

Comment:
@tadsul Can we make this a constant, describe it in javadocs and add a unit test? Thanks.